packages/rs-drive-abci/src/abci/handler/mod.rs

```diff
@@ -507,32 +525,24 @@ where
         } else {
             let block_execution_outcome::v0::BlockExecutionOutcome {
                 app_hash,
-                state_transition_results,
+                state_transitions_result: state_transition_results,
                 validator_set_update,
                 protocol_version,
             } = run_result.into_data().map_err(Error::Protocol)?;
 
             let platform_version = PlatformVersion::get(protocol_version)
                 .expect("must be set in run block proposer from existing platform version");
 
-            let mut invalid_tx_count = 0;
-            let mut valid_tx_count = 0;
+            let invalid_tx_count = state_transition_results.invalid_count();
+            let valid_tx_count = state_transition_results.valid_count();
 
             let tx_results = state_transition_results
+                .into_execution_results()
                 .into_iter()
-                .map(|(_, execution_result)| {
-                    let tx_result: ExecTxResult =
-                        execution_result.try_into_platform_versioned(platform_version)?;
-
-                    if tx_result.code == 0 {
-                        valid_tx_count += 1;
-                    } else {
-                        invalid_tx_count += 1;
-                    }
-
-                    Ok(tx_result)
+                .map(|execution_result| {
+                    execution_result.try_into_platform_versioned(platform_version)
```

Comment:
We need to  remove from results STs what we aren't expecting in the proposal based on state transition execution result (the same logic as in prepare proposal)